### PR TITLE
Update related items when [un]locked on another browser

### PR DESCRIPTION
**Pull request by pablopunk**
SDESK-4953

### CODE REVIEW

File: scripts/apps/relations/directives/RelatedItemsDirective.ts [hunk 1]
```diff
@@ -271,6 +283,15 @@ export function RelatedItemsDirective(authoringWorkspace: AuthoringWorkspaceServ
...
                 }
             });
+
+            const removeEventListeners = [
+                'item:lock',
+                'item:unlock',
+            ].map((_) => $rootScope.$on(_, scope.refreshRelatedItems));
+
+            scope.$on('$destroy', () => {
+                removeEventListeners.forEach((_) => _());
```
Comment (tomaskikutis) [inline scripts/apps/relations/directives/RelatedItemsDirective.ts]: What's up with the underscores? :D The convention is that underscores are used for argument names only when you don't use the argument.

Comment (pablopunk) [reply scripts/apps/relations/directives/RelatedItemsDirective.ts]: But it looks cool doesn't it? Haha I'll change it 👍

**Comment by tomaskikutis** [comment]
Why do you care about the lock? Shouldn't you only refresh related items every time they were changed? locking is only one type of change and in this case, it seems it wouldn't matter whether it was lock property changed or headline updated

**Comment by pablopunk** [comment]
@tomaskikutis hmmm well the task was only about the lock when someone opens it on another session, that's why I use the lock, not sure if it's necessary to listen to all update events.

**Comment by tomaskikutis** [comment]
Not to all, there should be a single event that fires no matter what changes. Might be `item:update` or `content:update`.

I know the task is about locking, but thinking of related issues is part of all tasks. What if status/headline changes? Would a user expect to see it? I'd say yes.

**Comment by pablopunk** [comment]
@tomaskikutis I included that case and I also included logic to not reload the items if the event doesn't belong to these related items

**Comment by tomaskikutis** [comment]
Yep, I was implicitly suggesting the same, to only reload changed items. Doesn't `item:update` or `content:update` fire when unlocking too? It would be better if one event was enough

**Comment by pablopunk** [comment]
I thought that too but it doesn't. None of both will trigger when locking nor unlocking

**Comment by tomaskikutis** [comment]
@petrjasek can we have a single event that fires whenever anything changes in the article? It's error-prone to have to deal with multiple events. You never know if you aren't missing some.

**Comment by petrjasek** [comment]
yes we can.. would still keep those specific ones but we can have another one which will be always sent as well

**Pull request merged**